ts_src/psbt.ts

```diff
@@ -747,16 +747,32 @@ function canFinalize(
       return hasSigs(1, input.partialSig);
     case 'multisig':
       const p2ms = payments.p2ms({ output: script });
-      return hasSigs(p2ms.m!, input.partialSig);
+      return hasSigs(p2ms.m!, input.partialSig, p2ms.pubkeys);
     default:
       return false;
   }
 }
 
-function hasSigs(neededSigs: number, partialSig?: any[]): boolean {
+function hasSigs(
+  neededSigs: number,
+  partialSig?: any[],
+  pubkeys?: Buffer[],
+): boolean {
   if (!partialSig) return false;
-  if (partialSig.length > neededSigs) throw new Error('Too many signatures');
-  return partialSig.length === neededSigs;
+  let sigs: any;
+  if (pubkeys) {
+    sigs = pubkeys
+      .map(pkey => {
+        const pubkey = ecPairFromPublicKey(pkey, { compressed: true })
+          .publicKey;
+        return partialSig.filter(pSig => pSig.pubkey.equals(pubkey))[0];
```

Comment:
I think `find` is a bit more appropriate in this use case.
```suggestion
        return partialSig.find(pSig => pSig.pubkey.equals(pubkey));
```